server/sonar-server/src/main/java/org/sonar/server/settings/ws/SetAction.java

```diff
@@ -112,12 +169,19 @@ private void checkPermissions(Optional<ComponentDto> component) {
   }
 
   private static SetRequest toWsRequest(Request request) {
-    return SetRequest.builder()
+    SetRequest setRequest = SetRequest.builder()
       .setKey(request.mandatoryParam(PARAM_KEY))
-      .setValue(request.mandatoryParam(PARAM_VALUE))
+      .setValue(request.param(PARAM_VALUE))
+      .setValues(request.multiParam(PARAM_VALUES))
       .setComponentId(request.param(PARAM_COMPONENT_ID))
       .setComponentKey(request.param(PARAM_COMPONENT_KEY))
       .build();
+
+    checkRequest((setRequest.getValue() != null && !setRequest.getValue().isEmpty())
+      ^ (setRequest.getValues() != null && !setRequest.getValues().isEmpty()),
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) NullPointerException might be thrown as 'getValues' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)